webapp/src/screens/tabs/UsersTable.tsx

```diff
@@ -0,0 +1,923 @@
+          <DialogContentText>
+            Please fill out the following fields to {user !== undefined ? "update the" : "create a new"} user.
+          </DialogContentText>
+          <TextField
+              autoFocus
+              margin="dense"
+              id="firstName"
+              fullWidth
+              variant="standard"
+              value={firstName ?? ''}
+              onChange={handleFirstNameChange}
+              error={firstNameError}
+              helperText={"First name"}
+          />
+          <TextField
+              autoFocus
+              margin="dense"
+              id="lastName"
+              fullWidth
+              variant="standard"
+              value={lastName ?? ''}
+              onChange={handleLastNameChange}
+              error={lastNameError}
+              helperText={"Last Name"}
+          />
+          <TextField
+              autoFocus
+              margin="dense"
+              id="emailAddress"
+              fullWidth
+              variant="standard"
+              value={emailAddress ?? ''}
+              onChange={handleEmailAddressChange}
+              error={emailAddressError}
+              helperText={"Email Address"}
+          />
+          <TextField
+              autoFocus
+              margin="dense"
+              id="username"
+              fullWidth
+              variant="standard"
+              value={username ?? ''}
+              onChange={handleUsernameChange}
+              error={usernameError}
+              helperText={"Username (if different from Email Address)"}
+          />
+          <TextField
+              autoFocus
+              margin="dense"
+              id="password"
+              fullWidth
+              variant="standard"
+              value={password ?? ''}
+              onChange={handlePasswordChange}
+              error={passwordError}
+              helperText={"Password"}
+          />
+          <TextField
+              autoFocus
+              margin="dense"
+              id="roleNames"
+              fullWidth
+              variant="standard"
+              value={roleNames ?? ''}
+              onChange={handleRoleNamesChange}
+              error={roleNamesError}
+              helperText={"User Roles (comma-separated)"}
+          />
+          <TextField
+              autoFocus
+              margin="dense"
+              id="courseIds"
+              fullWidth
+              variant="standard"
+              value={courseIds ?? ''}
+              onChange={handleCourseIdChange}
+              error={courseIdsError}
+              helperText={"Course IDs (comma-separated)"}
+          />
+          <TextField
+              autoFocus
+              margin="dense"
+              id="photoId"
+              fullWidth
+              variant="standard"
+              value={photoUrl ?? ''}
+              onChange={handlePhotoUrlChange}
+              error={photoUrlError}
+              helperText={"Profile Picture URL"}
```

Comment:
minor experience thing - i'd add an example photo url - like the Professor's photo url here